pkgs/applications/networking/cluster/ssm-session-manager-plugin/default.nix

```diff
@@ -5,9 +5,12 @@ let
     if stdenv.isDarwin then {
       url = "https://s3.amazonaws.com/session-manager-downloads/plugin/${ver}/mac/sessionmanager-bundle.zip";
       sha256 = "0gr6frdn9jvxnkymkcpvgkqw4z2sac9jdf5qj4hzakq1zkfviazf";
-    } else {
+    } else if stdenv.isx86_64 then {
```

Comment:
Please use attr select instead of if else chains.

Reply:
I can't find any reference to `attr select` in the docs, and grepping `nixpkgs` doesn't yield anything useful.

Can you point me to an example of what you have in mind? I can use that as a reference.

Reply:
Oh, I think [it's this](https://gist.github.com/joepie91/c3c047f3406aea9ec65eebce2ffd449d)?

I'm not sure how to write it for this particular case; there is not one branch per platform. There are four platforms, but three branches, since two share the same binaries.

Reply:
https://github.com/NixOS/nixpkgs/blob/master/pkgs/servers/prowlarr/default.nix#L12-L22

> There are four platforms, but three branches, since two share the same binaries.

This can be solved like

```nix
  hash = rec {
    x64-linux = "sha256-NTkb6E0GRofd63TozLPxyNZ6GxO4pAc+MWsUDKWLZDY=";
    arm64-linux = "sha256-eB1Izvf/E0iOGkWt6fSc4oAt1sJ7XTxn/iwDNYz7WkA=";
    x64-osx = "sha256-TAphdcbzbNUZpWYVaHBkhI+RsgxWmI8n8Z6VwYxHXWg=";
    arm64-osx = x64-osx;
  }."${arch}";
```